### DISCUSSION TITLE
Is there a way to make image pyramids for many images at once (batch conversion)?

### DISCUSSION THREAD

**Discussion by hhentschel** [libvips/libvips#2023] [Q&A]
Hello, 
i convert a large image to an image pyramid to use in the zoom-image viewer "openseadragon".
either with vips `dzsave xxx.jpg xxx` 
or if i have to alter it i use:
`vips icc_transform xxx.jpg xxx.dz srgb --intent perceptual --embedded`

Now i have about 100 pictures. Is there a way to change them all at once?
Like a batch processor?
I tried two images at once, but it gave me the error, "too many arguments".
Also, could not find anything in the docs.

**Comment by jcupitt** [comment]
You should use some kind of scripting language for this. For example, in bash you could write:

```bash
for filename in *.jpg; do
  vips dzsave $filename $(basename $filename .jpg)-pyr
done
```

**Comment by hhentschel** [comment]
Thanks for the hint.
But i dont really know, how to handle it.
I get the error: VipsForeignLoad: File  »*.jpg« not found.
I made a folder with all the images in it, the path in the terminal is in this folder.

also tried to put in the lines separately, but no luck.

**Comment by jcupitt** [comment]
You'd need to share details about your platform: what OS, what version, how it's configured, all that.

If you're dealing with lots of files, you'll find it very useful to learn a scripting language. Bash is probably the most widely used one on linux.

**Comment by hhentschel** [comment]
i am on macOSX Catalina.

i got it to work with the dzsave command. 
i put the script in an extra file, dzsave.sh
and in the terminal i type: bash dzsave.sh
(files and dzsave.sh have to be in the same directory.)

but now i tried it with icc_transform, but i cant get to the .dz ending into the $basename, 
while doing the loop.

what i am trying:
`for filename in *.jpeg; do
  vips icc_transform $filename $(basename $filename) srgb --intent perceptual --embedded
done`

side question: is that correct, that the ".dz" file is now outside the "*_files" folder,
after converting?
when i started with vips and dzsave some month ago, it was all in one folder.

**Comment by jcupitt** [comment]
Try something like:

```bash
vips icc_transform $filename $(basename $filename).dz srgb
```
